packages/react-components/src/text.ts

```diff
@@ -10,19 +11,26 @@ import {
 
 export type TextChild = React.ReactText | boolean | null | undefined;
 export type TextChildren = TextChild | ReadonlyArray<TextChild>;
+export type AllowedChild = React.ReactElement<HTMLStyleElement>;
```

Comment:
`TextChild` should be included in here already, not only in `AllowedChildren` (otherwise it's not simply the plural but actually different)